app/controllers/clinicfinders_controller.rb

```diff
@@ -4,20 +4,23 @@ class ClinicfindersController < ApplicationController
   def search
     return head :bad_request if params[:zip].blank?
 
-    # Clinics intentionally excluded from ClinicFinder are assigned the zip 99999.
-    # e.g. so a fund can have an 'OTHER CLINIC' catchall.
-    excluded_zip = '99999'
-
-    # Get all clinics except those with invalid zip codes.
-    clinic_finder = ClinicFinder::Locator.new(
-      Clinic.where(:zip.nin => [nil, '', excluded_zip]),
-      gestational_age: (params[:gestation].to_i || 0),
-      naf_only: params[:naf_only] == '1',
-      medicaid_only: params[:medicaid_only] == '1'
-    )
+    gestation = calculate_gestation params
+    filtered_clinics = Clinic.where(:zip.nin => [nil, '', Clinic::EXCLUDED_ZIP])
```

Comment:
this is a lot of logic for a controller - could move this set of wheres/filters to the Clinic model, or let it be here. Not sure if I have an opinion.

Reply:
Searches always flag like this for me. I don't have a strong opinion either.

Reply:
Let's leave it for now and extract if we decide we need to. My suspicion is that this won't get touched too often and we won't need to do this search elsewhere.